libs/ts/ui/src/components/DropdownMenu/DropdownMenu.tsx

```diff
@@ -146,10 +148,17 @@ export const DropdownMenuCheckboxItem = ({
 }: HTMLAttributes<HTMLDivElement> & {
   checked: boolean;
   onCheckedChange: (checked: boolean) => void;
-  children: ReactNode;
+  children: string;
 }) => {
-  const handleClick = () => {
-    onCheckedChange(!checked);
+  const handleClick = (event: MouseEvent<HTMLDivElement>) => {
+    if (event.target == event.currentTarget) {
+      event.currentTarget.querySelector('input')?.click();
+    }
+  };
+
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const { checked } = event.target;
+    onCheckedChange(checked);
   };
```

Comment:
```suggestion
  const [dropdownChecked, setDropdownChecked] = useState(checked);

  useEffect(() => {
    onCheckedChange(dropdownChecked);
  }, [dropdownChecked]);

  const handleClick = () => {
    setDropdownChecked(state => !state);
  };

  const handleChange = (e) => {
    setDropdownChecked(e.target.checked);
  }
```

It will be better to have controlled state.